django/db/migrations/operations/models.py

```diff
@@ -802,6 +805,7 @@ def state_forwards(self, app_label, state):
         model_state = state.models[app_label, self.model_name_lower]
         indexes = model_state.options[self.option_name]
         model_state.options[self.option_name] = [idx for idx in indexes if idx.name != self.name]
+        state.reload_model(app_label, self.model_name_lower, delay=True)
```

Comment:
I'll try to move the test to `test_state.py` and commit this patch without this untested line unless there's further feedback.